src/Analyzer.Utilities/JsonLineNumberResolver.cs

```diff
@@ -115,20 +117,108 @@ public int ResolveLineNumber(string pathInExpandedTemplate)
             var originalTemplatePath = tokenFromOriginalTemplate.Path;
             var unmatchedPathInOriginalTemplate = pathInExpandedTemplate[originalTemplatePath.Length..].TrimStart('.');
```

Comment:
This needs to update too (same problem with different index string lengths).  You can simply create an issue for this if you don't want to fix it now.